### PR TITLE
Add interop docs to API.md

### PR DESCRIPTION
**Pull request by paulosman**
- Adding the interop API functions to the API documentation
- Some general layout adjustments and adding a list of contents to ease findability

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -13,13 +13,53 @@ const beeline = require("honeycomb-beeline")();
...
+- [`withTimer()`](#withtimer)
+- [deprecated: `removeContext()`](#removecontext)
+- [deprecated: `customContext.add()`](#customcontextadd)
+- [deprecated: `customContext.remove()`](#customcontextremove)
+
 4.  [Async context bookkeeping](#async-context-bookkeeping)
 
+- [`bindFunctionToTrace()`](#bindfunctiontotrace)
+- [`runWithoutTrace()`](#runwithouttrace)
+- [`flush()`](#flush)
```
Comment (toshok) [inline docs/API.md:61]: `flush` isn't about async context.  it's about sending events to honeycomb.  Probably need a 5th section here ("Misc"?  ugh.)  or we can drop it for now.

**Review by toshok — APPROVED** [review]
going to approve-with-requests here since there's less potential for user impact.

**Comment by MikeGoldsmith** [comment]
@paul is this still wanted / needed? If yes, do you want to hand off to someone to fix up the final bits?

**Comment by vreynolds** [comment]
I removed a reference to `httpTraceParserHook` per suggestion, going to open another PR specifically around hooks usage.

**Review by MikeGoldsmith — APPROVED** [review]
Looks great, thanks for the updates @vreynolds 👍

**Pull request merged**